source/go-perun/app_tutorial/app.rst

```diff
@@ -0,0 +1,712 @@
+App
+...
+This section will implement the Tic-Tac-Toe game as a local channel app with a matching smart contract representation.
+
+.. note::
+    The smart contract will act as an on-chain reference to validate the game's state transitions.
+    This procedure is only needed in case of disputes.
+    The primary point of interaction is the local channel app.
+
+Data
+====
+Before we implement the channel app, we define the game's states.
+We put this in ``app/data.go``.
+
+The game state is handled in ``TicTacToeAppData``, which implements go-perun's ``Data`` interface.
+``NextActor`` represents the next party expected to make a move, and ``Grid`` represents the Tic-Tac-Toe 3x3 grid.
+The ``Grid`` indices are defined from the upper left to the lower right.
+
+.. code-block:: go
+
+    // TicTacToeAppData is the app data struct.
+    type TicTacToeAppData struct {  // Grid:
+        NextActor uint8             // 0 1 2
+        Grid      [9]FieldValue     // 3 4 5
+    }                               // 6 7 8
+
+
+Encode
+~~~~~~
+The ``Data`` interface requires that ``TicTacToeData`` can be encoded onto an ``io.Writer`` via ``Encode``.
+This method writes our game data to the given data stream.
+It is needed to push the local game data to the smart contract.
+Decoding will take place in the :ref:`app implementation <app_decoding>`.
+
+.. code-block:: go
```

Comment:
Please replace all code blocks with references to the files.

Reply:
I will do this once the actual app code is finalized & published. Until then lines of code can change and mess up the `literalinclude`